src/utils/api.js

```diff
@@ -20,14 +39,49 @@ let newListing = {
   condition: "",
   location: "",
   tradeRequired: false,
-  user_id: ""
+  user_id: "",
 };
 
+let listingRef;
 export const postListing = async (event, newListing) => {
   event.preventDefault();
-
   const docRef = await addDoc(collection(db, "listings"), newListing);
   console.log("Document written to Listings with ID: ", docRef.id);
+  listingRef = docRef.id;
+  uploadImage(event);
+  return listingRef;
+};
+
+const storageRef = ref_storage(storage);
+const metadata = {
+  contentType: "image/jpeg",
+};
+
+export const uploadImage = (event) => {
+  event.preventDefault();
+  const file = event.target[6].files[0];
+  if (!file) return;
+  const storageRef = ref_storage(
+    storage,
+    `/files/${uid}/${listingRef}/${file}`
+  );
+  const uploadTask = uploadBytesResumable(storageRef, file, metadata)
+    .then((snapshot) => {
+      return snapshot;
```

Comment:
is this snapshot data used anywhere? Looks like this .then/return is redundant and could be removed and just go into the next one? ie, just provide snapshot to getDownLoadURL function